File: Data Structures & Algorithms/two-integer-sum/submission-5.rs
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut store = HashMap::new();
        let n = nums.len();

        for (i, &num) in nums.iter().enumerate() {
            let diff = target - num;

            if let Some(&prev_index) = store.get(&diff) {
                return  vec![prev_index, i as i32]
            }
            else{
                store.insert(num, i as i32);
            }
        }
        vec![]
}
}

File: Data Structures & Algorithms/search-2d-matrix/submission-5.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l,r = 0,len(matrix)-1

        while l<=r:
            m = r+l//2
            if matrix[m][-1] < target:
                l=m+1
            elif matrix[m][0] > target:
                r=m-1
            else:
                return target in matrix[m]
        return False
            


